framework/src/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -942,10 +942,18 @@ public AnnotatedTypeMirror getAnnotatedType(Tree tree) {
             return treeCache.get(tree).deepCopy();
         }
 
-        AnnotatedTypeMirror type;
         if (TreeUtils.isClassTree(tree)) {
-            type = fromClass((ClassTree)tree);
-        } else if (tree.getKind() == Tree.Kind.METHOD ||
+            ClassTree classTree = (ClassTree) tree;
+            AnnotatedTypeMirror type = getAnnotatedTypeOfClassTree(classTree);
+            if (shouldCache) {
```

Comment:
Because if the type is cached when getAnnotatedTypeOfClassTree is called from a different method, then postProcessClassTree would not be called for that tree.